pkg/api/v1beta3/types.go

```diff
@@ -781,7 +781,7 @@ type PodSpec struct {
 	// Optional: Set DNS policy.  Defaults to "ClusterFirst"
 	DNSPolicy DNSPolicy `json:"dnsPolicy,omitempty" description:"DNS policy for containers within the pod; one of 'ClusterFirst' or 'Default'"`
 	// NodeSelector is a selector which must be true for the pod to fit on a node
-	NodeSelector map[string]string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node"`
+	NodeSelector string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node"`
```

Comment:
The question is whether this is a very important fundamental change in order for us to allow true selectors.  Agree the bar is high - Brian should weigh in.

----- Original Message -----
> > @@ -781,7 +781,7 @@ type PodSpec struct {
> >  	// Optional: Set DNS policy.  Defaults to "ClusterFirst"
> >  	DNSPolicy DNSPolicy `json:"dnsPolicy,omitempty" description:"DNS policy
> >  	for containers within the pod; one of 'ClusterFirst' or 'Default'"`
> >  	// NodeSelector is a selector which must be true for the pod to fit on a
> >  	node
> > -	NodeSelector map[string]string `json:"nodeSelector,omitempty"
> > description:"selector which must match a node's labels for the pod to be
> > scheduled on that node"`
> > +	NodeSelector string `json:"nodeSelector,omitempty" description:"selector
> > which must match a node's labels for the pod to be scheduled on that
> > node"`
> 
> v1b3 is out and enabled.  You can petition for a breaking change between
> v1beta3 and v1, but that is a pretty high bar.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7053/files#r28714176